src/storage/engine/mod.rs

```diff
@@ -62,6 +62,39 @@ pub trait Cursor {
     fn key(&self) -> &[u8];
     fn value(&self) -> &[u8];
 
+    /// Seek the specified key.
+    ///
+    /// This method assume the current position of cursor is
+    /// around `key`, otherwise you should use `seek` instead.
+    fn near_seek(&mut self, key: &Key) -> Result<bool> {
+        if !self.valid() {
+            return self.seek(key);
+        }
+        if self.key() == &**key.encoded() {
+            return Ok(true);
+        }
+        while self.key() > &**key.encoded() && self.prev() {}
```

Comment:
Can we guarantee the key is near?
If not, maybe we should add a max count check, otherwise, we may scan too many rows.